pkg/types/metadata.go

```diff
@@ -77,15 +77,20 @@ func (m *Metadata) UnmarshalJSON(data []byte) error {
 }
 
 func (m *Metadata) ToRego() interface{} {
-	return map[string]interface{}{
-		"filepath":  m.Range().GetFilename(),
-		"startline": m.Range().GetStartLine(),
-		"endline":   m.Range().GetEndLine(),
-		"managed":   m.isManaged,
-		"explicit":  m.isExplicit,
-		"fskey":     CreateFSKey(m.Range().GetFS()),
-		"resource":  m.Reference(),
+	input := map[string]interface{}{
+		"filepath":     m.Range().GetLocalFilename(),
+		"startline":    m.Range().GetStartLine(),
+		"endline":      m.Range().GetEndLine(),
+		"sourceprefix": m.Range().GetSourcePrefix(),
+		"managed":      m.isManaged,
+		"explicit":     m.isExplicit,
+		"fskey":        CreateFSKey(m.Range().GetFS()),
+		"resource":     m.Reference(),
+	}
+	if m.parent != nil {
+		input["parent"] = m.parent.ToRego()
 	}
+	return input
```

Comment:
I see you've refactored this and used it in all the `ToRego()` calls that are elsewhere, which is fantastic. Could you add a basic unit test that exercises this? 

So basically `metadata_test.go` to go along with this that can exercise the various calls. I would create a `NewMetadata` and then simply call `ToRego` on it to observe and assert the behaviour.

Reply:
That way you won't have to test every single type (string, int, etc.) individually.

Reply:
added a test that exercises the non-parent and parent use-cases